cactus/page.py

```diff
@@ -93,15 +88,8 @@ def render(self):
         Takes the template data with context and renders it to the final output file.
         """
 
-        data = self.data()
-        context = self.context(data=data)
-
-        # This is not very nice, but we already used the header context in the
-        # page context, so we don't need it anymore.
-        page_context, data = self.parse_context(data)
-
         context, data = self.site.plugin_manager.preBuildPage(
-            self.site, self, context, data)
+            self.site, self, self.context(data=self.data()), self.data())
```

Comment:
See note above — we might as well remove `self.data()` here, since we don't use it.